src/xdocs/config_filters.xml

```diff
@@ -648,6 +648,220 @@ public class UserService {
+            event.  Optional as long as <code>checks</code> or <code>message</code> is specified.
+          </li>
+          <li>
+            <code>query</code> -
+            a <a href="property_types.html#string">string</a>
+            xpath query. It is optional.
+          </li>
+        </ul>
+        <p>
+          Each audit event is checked against
+          each <code>suppress</code> and <code>suppress-xpath</code> element. It is
+          suppressed if all specified attributes match against the audit
+          event.
+        </p>
+        <p>
+          The following suppressions XML document directs
+          a <code>SuppressionXpathFilter</code> to
+          reject <code>CyclomaticComplexity</code> errors for
+          all methods with name <i>sayHelloWorld</i> inside <i>FileOne</i>
+          and <i>FileTwo</i> classes:
+        </p>
+        <source>
+&lt;?xml version=&quot;1.0&quot;?&gt;
+
+&lt;!DOCTYPE suppressions PUBLIC
+&quot;-//Puppy Crawl//DTD Suppressions Xpath Experimental 1.2//EN&quot;
+&quot;http://checkstyle.sourceforge.net/dtds/suppressions_1_2_xpath_experimental.dtd&quot;&gt;
+
+&lt;suppressions&gt;
+  &lt;suppress-xpath checks=&quot;CyclomaticComplexity&quot;
+  files=&quot;FileOne.java,FileTwo.java&quot;
+  query=&quot;//METHOD_DEF[@text='sayHelloWorld']&quot;/&gt;
+&lt;/suppressions&gt;
+        </source>
+        <p>
+          Suppress checks for package definitions:
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;.*&quot; query=&quot;/PACKAGE_DEF&quot;/&gt;
+        </source>
+        <p>
+          Suppress checks for parent element of the first variable definition:
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;.*&quot; query=&quot;(//VARIABLE_DEF)[1]/..&quot;/&gt;
+        </source>
+        <p>
+          Suppress checks for elements which are either class definitions,
+          either method definitions.
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;.*&quot; query=&quot;//CLASS_DEF | //METHOD_DEF&quot;/&gt;
+        </source>
+        <p>
+          Suppress checks for certain methods:
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;.*&quot; query=&quot;//METHOD_DEF[@text='getSomeVar'
+          or @text='setSomeVar']&quot;/&gt;
+        </source>
+        <p>
+          Suppress checks for variable <i>testVariable</i> inside <i>testMethod</i> method
+          inside <i>TestClass</i> class.
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;.*&quot; query=&quot;/CLASS_DEF[@text='TestClass']
+          //METHOD_DEF[@text='testMethod']//VARIABLE_DEF[@text='testVariable']&quot;/&gt;
+        </source>
+      </subsection>
+      <subsection name="Example of Usage">
+        <ul>
+          <li>
+            <a href="https://github.com/search?q=path%3Aconfig+filename%3Acheckstyle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+SuppressionXpathFilter">
+              Checkstyle Style</a>
```

Comment:
link to config/suppressions-xpath.xml might very useful too, as xpath creation will be most complicated part of Filter.

Reply:
```
     <subsection name="Example of Usage">
        <ul>
          <li>
            <a href="https://github.com/search?q=path%3Aconfig+filename%3Acheckstyle_checks.xml+repo%3Acheckstyle%2Fcheckstyle+SuppressionXpathFilter">
              Checkstyle Style</a>
          </li>
          <li>
            <a href="https://github.com/checkstyle/checkstyle/blob/master/config/suppressions-xpath.xml">
              Xpath-suppressions config file</a>
          </li>
        </ul>
      </subsection>
```
I tried to do this way, but `XdocsPagesTest#testAllCheckSections` fails, since only `Checkstyle Style`, `Google Style` and `Sun Style` texts are allowed inside `Example of Usage` section.
Should I modify test to accept `SuppressionXpathFilter` section?

Reply:
please create separate issue on this , we will discuss it later with @rnveach .
It is ok to skip this point from todo in this PR.

Reply:
done